### PR TITLE
Kafka 3.8.0 compatibility

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Enhancement / new feature

### Description

Changes to achieve Kafka 3.8.0 compatibility made in manner to retain compatibility with earlier versions.
(Tested against 3.8.0 RC1)

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by k-wall** [comment]
I think it would be useful to merge/release this soon to help those who are testing their component at the 3.8.0 RCs.

File: impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java [hunk 1]
```diff
@@ -168,7 +168,7 @@ private KafkaConfig buildBrokerConfig(KafkaClusterConfig.ConfigHolder c) {
         Properties properties = new Properties();
         properties.putAll(c.getProperties());
         var logsDir = getBrokerLogDir(c.getBrokerNum());
-        properties.setProperty(KafkaConfig.LogDirProp(), logsDir.toAbsolutePath().toString());
+        properties.setProperty("log.dir", logsDir.toAbsolutePath().toString());
```
Comment (SamBarker) [inline impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java:171]: again feels odd to be moving towards hardcoded strings from lookups.

File: impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java [hunk 2]
```diff
@@ -516,7 +515,7 @@ private void configureKraftNode(KafkaEndpoints kafkaEndpoints, int nodeId, Prope
         putConfig(nodeConfiguration, "node.id", Integer.toString(nodeId)); // Required by Kafka 3.3 onwards.
 
         var quorumVoters = IntStream.range(0, kraftControllers)
-                .mapToObj(controllerId -> String.format("%d@//%s", controllerId, kafkaEndpoints.getEndpointPair(Listener.CONTROLLER, controllerId).connectAddress()))
+                .mapToObj(controllerId -> String.format("%d@%s", controllerId, kafkaEndpoints.getEndpointPair(Listener.CONTROLLER, controllerId).connectAddress()))
```
Comment (SamBarker) [inline impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java:518]: Was the format wrong before or did it change as part of 3.8.0? I guess what I'm really getting at is does this need to checked per version somehow.

**Review by SamBarker — COMMENTED** [review]
@k-wall  I don't really understand why this helps compatibility? Moving from property lookups to hard coded names feels like moving in the wrong direction. 

If it gives us compatibility across Kafka versions then I'm +1 to merging I just don't follow as things stand.

**Comment by k-wall** [comment]
The property methods have been removed in Kafka 3.8.0. Replacements exist in other classes, but I couldn't see a reasonable why to utilise them and retain compile time compatibility.

I want to merge this before 3.8.0 is release so that users (Strimzi) have a release that works against new and old.

File: impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java (hunk 2 above)
Comment (k-wall) [reply impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java:518]: The format used by the Ext for this property was wrong (according to the Kafka documentation).  The validation used by 3.8.0 rejects the slashes.

**Review by SamBarker — APPROVED** [review]
Given the lack of compile time property lookups this LGTM

**Pull request merged**